app/helpers/govuk_list_helper.rb

```diff
@@ -0,0 +1,36 @@
+module GovukListHelper
+  def govuk_list(array = nil, type: nil, spaced: nil, classes: nil, html_attributes: {}, &block)
+    html_classes = ["#{brand}-list"]
+
+    case type.to_s.to_sym
+    when :bullet
+      html_classes << ["#{brand}-list--bullet"]
+      tag_type = "ul"
+    when :number
+      html_classes << ["#{brand}-list--number"]
+      tag_type = "ol"
+    when :""
+      tag_type = "ul"
+    else
+      raise "Unrecognised type for govuk_list - should be :bullet or :number or nil"
+    end
+
+    if spaced
+      html_classes << ["#{brand}-list--spaced"]
+    end
+
+    html_classes += Array.wrap(classes)
```

Comment:
I think Rails' `#class_names` can make this a bit clearer, something along the lines of:

```suggestion
    type = type.to_s
    fail "Unrecognised type for govuk_list - should be :bullet or :number or nil" unless type.in?(["bullet", "number", ""])

    tag_type = (type == "number") ? "ol" : "ul"

    html_classes = class_names(
      "#{brand}-list",
      *classes,
      "#{brand}-list--bullet" => type.to_s == "bullet",
      "#{brand}-list--number" => type.to_s == "number",
      "#{brand}-list--spaced" => spaced
    )
```

Reply:
I’d not come across `class_name` before.

I had a quick go at this, but I think the custom classes (`*classes`) ought to appear after the `govuk-` ones as it can affect the CSS if they otherwise conflict, and `*classes` can't appear after the conditional ones for some reason.

I’ve used your suggestion to refactor the rest of this a bit though.

Reply:
Yeah, the arguments for class names is basically `*args, **kwargs` so you can't add an extra one on the end. CSS class order doesn't matter though - I think the worst that can happen is that someone might do `type: "number"` **and** `classes: "govuk-list--bullet"`, which might be confusing.

Reply:
It's the order in the CSS that matters though, not the HTML, like here the second item should be yellow:

https://codepen.io/peteryates/pen/mdgerEQ

Reply:
You’re right, I remembered that completely wrong! 🤦‍♂️

Reply:
I've switched to using `class_names` now. 👍  One advantage it has is avoiding using `Array.wrap()` which is a Rails monkeypatch. Makes it easier if we were able to find a way to avoid the Rails dependency in future.